reprovider/interface.md

```diff
@@ -0,0 +1,53 @@
+# Provider Interface
+
+## Suggested `SweepingReprovider` interface
+
+```go
+type Provider interface {
```

Comment:
This interface is currently named `Reprovider`, in `go-libp2p-kad-dht/reprovider`.

I figured the name `Provider` makes more sense since its users only need to know that this is a content routing provider. They don't need to know the internals of whether the content has to be reprovided. The goal is to signal that the `Provider` takes the responsibility to advertise content as needed to the remote content routing system (DHT here).

I initially chose `Reprovider` because `go-libp2p-kad-dht/providers/` already exists, and because kubo Provider is the old boxo providing system. I now think it makes more sense to use `Provider`.